### PR TITLE
Refactor MerchStoreService + Misc. Bug Fixes

### PR DESCRIPTION
**Pull request by nik-dange**
# Info

Closes **#200**. 

# Description

The MerchStoreService contains logic for both Merch Store functionality as well as Merch Order functionality. Refactoring into a `MerchStoreService` and a `MerchOrderService` for ease of development moving forward. Also random bug fixes.

## Changes

- Created a new `MerchOrderService`
- Fixed a bug allowing members to reschedule order pickup to a full event
- Fixed a bug preventing merch item collections from being edited

# Type of Change

- [ ] Patch (non-breaking change/bugfix)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Documentation (A change to a README/description)
- [ ] Continuous Integration/DevOps Change (Related to deployment steps, continuous integration
      workflows, linting, etc.)
- [ ] Other: (Fill In) 

If you've selected Patch, Minor, or Major as your change type, **make sure to bump the version before merging in `package.json`!**
# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally.
- [ ] on the testing API/testing database.
- [ ] with appropriate Postman routes. Screenshots are included below.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [ ] I have appropriately edited the API version in the `package.json` file.
- [x] My changes produce no new warnings.

# Screenshots

Please include a screenshot of your Postman testing passing successfully.

### CODE REVIEW

File: services/MerchStoreService.ts [hunk 1]
```diff
@@ -47,13 +35,39 @@ export default class MerchStoreService {
...
-
   private transactions: TransactionsManager;
 
-  constructor(@InjectManager() entityManager: EntityManager, emailService: EmailService) {
+  constructor(@InjectManager() entityManager: EntityManager) {
     this.transactions = new TransactionsManager(entityManager);
-    this.emailService = emailService;
+  }
+
+  public async findItemByUuid(uuid: Uuid, user: UserModel): Promise<PublicMerchItemWithPurchaseLimits> {
```
Comment (maxwn04) [inline services/MerchStoreService.ts:44]: Inconsistent external vs internal types, service returning a public type

**Review by maxwn04 — COMMENTED** [review]
Looks good to me. Kinda just skimmed the changes since its mostly just moving between files and focused on the bug fixes.

**Review by dowhep — APPROVED** [review]
did not break portal

LGTM 🚀 🚀

**Review by jennymar — APPROVED** [review]
Tested locally on Postman - LGTM!

**Pull request merged**